### ISSUE TITLE
version 2.2.0  filling logs with unnecessary messages

### ISSUE THREAD

**Issue by loudgazelle** [bwp91/homebridge-platform-wemo#214]
I upgraded to v 2.2.0 this morning and the logs immediately began filling with messages saying `[bulb name] not found yet so cannot be controlled. Will retry connection.` - this happened for each of my bulbs and would repeat every few minutes. This informational messages repeatedly being written clutters the logs and drowns out any other issues that might need to be addressed, so I almost immediately reverted to version 2.1.2. 

A sample of the logs are below. It's notable that despite these messages I was able to successfully turn on/off a lamp several times starting at `11:12:36 AM`, so I'm not sure what the point of the messages is. 

Aside from the Netcam devices (which aren't connected any longer, but that's another story I won't get into now), all of the devices it's complaining about were connected via a [Wemo Link](http://belkin.force.com/Articles/articles/en_US/How_To/Setting-up-the-WEMO-LED-Lighting-Starter-Set) device. The Wemo Insight I have also connects to HomeKit through homebridge-platform-wemo but was not mentioned in the logs. 

I suppose it might be useful to _someone_ for these log entries to exist or you wouldn't have added them, so as an alternative to removing them completely it'd be nice if there was a setting to control whether they show up - especially if the default was for the messages to not show up so that a user would have to explicitly turn them on. 

https://github.com/bwp91/homebridge-platform-wemo/blob/83758a4dd0adbc5bbca5887257ed874b17b4a9d4/lib/wemo-platform.js#L133-L149

```
[11/11/2020, 10:39:32 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:39:32 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:39:32 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:39:32 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:42:03 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:44:33 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:47:03 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:49:33 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:52:04 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:54:34 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:57:04 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 10:59:35 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:02:05 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:04:36 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:07:06 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:09:36 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:07 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:12:36 AM] [Belkin WeMo] [Chair Lamp] setting state to [on].
[11/11/2020, 11:12:44 AM] [Belkin WeMo] [Chair Lamp] setting state to [off].
[11/11/2020, 11:12:45 AM] [Belkin WeMo] [Chair Lamp] updating state to [on].
[11/11/2020, 11:12:49 AM] [Belkin WeMo] [Chair Lamp] updating state to [off].
[11/11/2020, 11:12:52 AM] [Belkin WeMo] [Chair Lamp] updating state to [on].
[11/11/2020, 11:12:54 AM] [Belkin WeMo] [Chair Lamp] updating state to [off].
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:14:37 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [Arthur’s netcam] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [Iggy’s Netcam ] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [Chair Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [French Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [leg lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [Tall Lamp] not found yet so cannot be controlled. Will retry connection.
[11/11/2020, 11:17:07 AM] [Belkin WeMo] [Lightbulb 01] not found yet so cannot be controlled. Will retry connection.
```

**Comment by bwp91** [comment]
Hi @loudgazelle thanks for bringing this to my attention.

These `[device x] not found yet so cannot be controlled` are meant to be informational but in most cases should be limited to a short period after Homebridge starts. Once HB has matched every accessory in the cache with a discovered device, there should be no more repeated messages. The log messages start 30 seconds after HB has finished loading, to give the plugin an initial change to discover devices without *every* device appearing in the logs. This is why your there was no mention of your Insight in the logs.

I realise now the repeated messages about devices connected via the Wemo Link is a code error that I have overlooked. If you have the time, would you be willing to try a beta version (not published yet) to see if it fixes this? (As I don't own the Wemo Link myself)

The repeated messages about the NetCams will be there because you still have the motion sensors in Homebridge, and Homebridge is trying to "match" them to a device discovered on your network. When you mentioned they aren't connected anymore, I assume you mean unplugged ie not in use at all? If the devices are never found then these log messages will indeed go on repeating. If you no longer use the NetCams, perhaps it would be worth removing the motion sensor accessories from the Homebridge cache?

**Comment by loudgazelle** [comment]
Sure, I'm willing to test a beta version. 

> The repeated messages about the NetCams will be there because you still have the motion sensors in Homebridge, and Homebridge is trying to "match" them to a device discovered on your network. When you mentioned they aren't connected anymore, I assume you mean unplugged ie not in use at all? If the devices are never found then these log messages will indeed go on repeating. If you no longer use the NetCams, perhaps it would be worth removing the motion sensor accessories from the Homebridge cache?

This is something I'd been struggling with for a while. I had added the serial numbers for both netcams to the `ignoredDevices` section of the config but they were still showing up. Today I manually removed them from the `cachedAccessories` file but when HomeBridge restarted it would add them back. Finally I was able to remove them by shutting down homebridge completely, removing them from that file, then starting homebridge back up.

**Comment by bwp91** [comment]
> Sure, I'm willing to test a beta version.

Great, I've just published v2.3.0-2. The method of updating differs depending on how/which platform you're running Homebridge on, but maybe you'd know how to do this already? Let me know if you need any help.

I've added a fix for the wemo link, so the relentless logging should no longer happen once the link has been discovered!

**Comment by loudgazelle** [comment]
v2.3.0-2 seems to have resolved the issue I originally reported in this thread - thanks for the quick turnaround!

Unfortunately, it introduced another issue - at least, I've never seen this issue prior to this version. 

As you can see in the logs below the logs are now filling up with wemo insight updates about the on-time and total consumption.  

```
[11/11/2020, 5:55:06 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:06 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:07 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:07 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:10 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:10 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:11 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:11 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:14 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:14 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:15 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:15 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:18 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:18 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:19 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:19 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:22 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:22 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:23 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:23 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:26 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.05 hours.
[11/11/2020, 5:55:26 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:27 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:27 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:30 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:30 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:31 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:31 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:34 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:34 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:35 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:35 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:38 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:38 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:39 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:39 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:42 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:42 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:43 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:43 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:47 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:47 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
[11/11/2020, 5:55:48 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 11.06 hours.
[11/11/2020, 5:55:48 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.07 kWh.
```

prior to installing v2.3.0-2 this data would be logged about every 10 minutes, which is much more tolerable. 
```
[11/11/2020, 4:47:38 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 9.92 hours.
[11/11/2020, 4:47:38 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.063 kWh.
[11/11/2020, 4:57:51 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 10.1 hours.
[11/11/2020, 4:57:51 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.064 kWh.
[11/11/2020, 5:07:03 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 10.25 hours.
[11/11/2020, 5:07:03 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.065 kWh.
[11/11/2020, 5:17:16 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 10.42 hours.
[11/11/2020, 5:17:16 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.066 kWh.
[11/11/2020, 5:26:28 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 10.57 hours.
[11/11/2020, 5:26:28 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.067 kWh.
[11/11/2020, 5:36:42 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 10.74 hours.
[11/11/2020, 5:36:42 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.068 kWh.
[11/11/2020, 5:45:53 PM] [Belkin WeMo] [Mission Lamp] updating total on-time - 10.9 hours.
[11/11/2020, 5:45:53 PM] [Belkin WeMo] [Mission Lamp] updating total consumption - 0.069 kWh.
```

for the time being I've reverted to 2.1.2 again, but I'm willing to continue testing to help isolate this new issue.

**Comment by bwp91** [comment]
New beta, I hope v2.3.0-3 will fix the consumption repeated messages!

**Comment by loudgazelle** [comment]
the only beta version that homebridge ui x showed was 2.3.0-5 so I tried that version but the consumption messages were still coming in every second or two, so I've reverted again to 2.1.2.

**Comment by bwp91** [comment]
v2.3.0-6 should definitely now stop the messages. I appreciate your patience and feedback on this 👍

**Comment by benfysh** [comment]
Hi, I am using v2.3.1-7 and I am still seeing excessive logging. The lights listed is the below extract are working as expected. Thanks

```
[11/16/2020, 13:44:21] [WeMo Platform] [Garage Right] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:44:21] [WeMo Platform] [Dining room] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:44:21] [WeMo Platform] [Garage Left] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:46:51] [WeMo Platform] [Garage Right] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:46:51] [WeMo Platform] [Dining room] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:46:51] [WeMo Platform] [Garage Left] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:49:21] [WeMo Platform] [Garage Right] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:49:21] [WeMo Platform] [Dining room] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:49:21] [WeMo Platform] [Garage Left] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:51:51] [WeMo Platform] [Garage Right] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:51:51] [WeMo Platform] [Dining room] not found yet so cannot be controlled. Will retry connection.
[11/16/2020, 13:51:51] [WeMo Platform] [Garage Left] not found yet so cannot be controlled. Will retry connection.
```

**Comment by bwp91** [comment]
Hi @benfysh could you try v2.3.1-7 beta to see if this is fixed? thanks

**Comment by loudgazelle** [comment]
I'm trying to move to v2.3.1-8 and Homebridge-ui-x is warning me that the recommended node.js version has changed: 

> The homebridge-platform-wemo plugin recommends running on Node.js v14.5.0 or later. You are currently running Node.js v12.19.0.
> 
> Are you sure you want to update homebridge-platform-wemo?

I'm running 12.19.0 and leery of updating to 14.5.0 because of my past experience updating to new major versions of node and the amount of rework I have had to go through getting homebridge working correctly again. Maybe my approach to those upgrades isn't right, though. 

is node 14.5 required, or will it work on node 12.19?

**Comment by benfysh** [comment]
hi @bwp91 that seems to have done the trick, Thanks. 

@loudgazelle it seems to be working fine on node 12.19 for me.

**Comment by loudgazelle** [comment]
thanks @benfysh 

v 2.3.1-8 has resolved the excessive messaging about power consumption from the Wemo Insight - thanks, I really appreciate your focus on this and my previous issue!

**Comment by bwp91** [comment]
Glad all is well with issues with the logs.

@loudgazelle this plugin will still work fine with node v12. The warning you see is just a warning, not an error. I've chosen to put this message there to make people aware there there is a newer version of node.js who otherwise wouldn't realise.

**Comment by bwp91** [comment]
Wemo Link and Wemo Insight consumption repeated logging fixes have been released in v2.3.1 :)

**Comment by idcrook** [comment]
> 
> This is something I'd been struggling with for a while. I had added the serial numbers for both netcams to the `ignoredDevices` section of the config but they were still showing up. Today I manually removed them from the `cachedAccessories` file but when HomeBridge restarted it would add them back. Finally I was able to remove them by shutting down homebridge completely, removing them from that file, then starting homebridge back up.

@bwp91 If a device serial is listed in `ignoredDevices`, it is still producing `[] not found yet so cannot be controlled. Will retry connection.` messages.  I would expect that putting a device in `ignoredDevices` list, that it would be rejected from cached accessories, etc.  

Should I file another issue for this?

**Comment by bwp91** [comment]
Hi @idcrook no need to file another issue.

In this case, the log messages were appearing as you already had the device in the Homebridge cache. The discovery of this device was being ignored, but the plugin was still waiting to match the device in the cache with the device on the network which it would never find (as ignored), leading to this endless loop.

Would you be willing to try a beta, I've slightly refactored how the plugin deals with `ignoredDevices`. The beta in question is v2.4.1-0. For each device discovered, the plugin will now check to see if the device is in the Homebridge cache, remove it, *then* check if it is an ignored device and skip if if necessary (so it is not added again).

**Comment by idcrook** [comment]
> Would you be willing to try a beta, I've slightly refactored how the plugin deals with `ignoredDevices`. The beta in question is v2.4.1-1. For each device discovered, the plugin will now check to see if the device is in the Homebridge cache, remove it, _then_ check if it is an ignored device and skip if if necessary (so it is not added again).

Thanks.  will try to take a look later today

**Comment by idcrook** [comment]
Ok, I had a chance to try out.  I use nvm on macOS

```console
$ npm install -g 'bwp91/homebridge-platform-wemo#latest'
$ npm -g --depth=0 list | grep homebridge-platform-wemo
├── homebridge-platform-wemo@2.4.1-2
```

devices in cached list **are still** periodically getting the message `11/19/2020, 9:10:03 AM] [WeMo Platform] [<<NAME>>] not found yet so cannot be controlled. Will retry connection.` even though their serial number is listed in `ignoredDevices` list

Devices in the `ignoredDevices` list  are *not* appearing in messages the initial search for devices `11/19/2020, 9:04:32 AM] [WeMo Platform] Plugin [v2.4.1-2] initialised. Searching for devices...` (not sure if this behavior changed but thought I'd point it out). 

The `not yet found` messages are only starting to appear after `Homebridge is running on port 62002` message in log file

**Comment by bwp91** [comment]
@idcrook which wemo devices do you own, any in particular that still seem to be causing these `cannot be controlled`messages?

**Comment by idcrook** [comment]
> @idcrook which wemo devices do you own, any in particular that still seem to be causing these `cannot be controlled`messages?

This is happening for Insight Switch, Switch, and Motion devices that have been included in ignoredDevices list. It should be reproducible by simply unplugging the device (and having serial number in `ignoredDevices` list).  
I used the cached devices JSON file itself to determine their serial numbers.

**Comment by bwp91** [comment]
I'm afraid I cannot recreate this, I have added my own wemo switch's serial number to my ignoredDevices list, restarted HB and it was removed from HB with no further log messages.
Could you post your wemo config here?

**Comment by idcrook** [comment]
It's a pretty vanilla config for WeMo, but for the `ignoredDevices`

```
 {
      "platform": "BelkinWeMo",
      "name": "WeMo Platform",
      "noMotionTimer": 60,
      "ignoredDevices": ["22REDACTED", <<TRUNCATED>>]
    },
```

I unplug some devices for long periods, especially at different times of the year, which is why I wish for them to be ignored (when thus configured). I would be happy to create a new issue for this.

To reproduce:
 - homebridge reporting device not found (because device was not plugged into mains, etc.); shutdown homebridge
 - device is listed in the cached accessories list on disk.
 - device is unplugged from mains, include its serial number in `ignoredDevices`
 - start homebridge
 - wemo plugin still seems to be searching for device, even if in `ignoredDevices`

**Comment by bwp91** [comment]
@idcrook I understand now. And I can see how you're facing problems from my perspective knowing how the plugin works.
The plugin checks whether a device is in the ignoreDevices list **on discovery**. I wasn't under the impression in your case (nor had even thought of the case) that a user would purposefully have a device unplugged.

Looks like I need to implement this check (and removal of device from HB) at plugin runtime rather than on discovery. I'll get back to you with another beta to try!

**Comment by bwp91** [comment]
reopening

**Comment by bwp91** [comment]
Okay, new beta, v2.4.1-3.
The change is that the plugin will now remove accessories from HB that are in `ignoredDevices` on plugin startup rather than on device discovery.
A slight caveat is that, as a one-off per device, it will need to be discovered by the plugin for these changes to take effect. Until it is discovered for the first time, the `cannot be controlled` log entries will continue.

**Comment by idcrook** [comment]
The latest does not change the undesired behavior. 

```console
$ npm -g --depth=0 list | grep homebridge-platform-wemo
├── homebridge-platform-wemo@2.4.1-5
```

ignoredDevices list devices are still showing up periodically in `cannot be controlled` log messages 

Also, in your explanation, how could an unpowered device be "discovered"? I think there is still a missing piece here, as these devices just happened to be in the cached devices list.

If a device is explicitly in the `ignoredDevices` list, why would they continue to be looked for in discovery? Maybe I am misunderstanding?

**Comment by bwp91** [comment]
The changes won't take effect until each device in your `ignoredDevices` list has been discovered by the plugin once. As you have said, an unpowered device cannot be discovered. So as a one off for each device, it will need to be discovered by the plugin by being:

* powered on
* not in the `ignoredDevices` list

Then if you add the device back into `ignoredDevices`, it will be removed from the Homebridge cache on Homebridge restart, so won't be looked for, so no logging about it.

The other option as a one off is to manually remove these devices from the Homebridge cache. This might be a better option if you, for example, are unable to plug them in.

Once this is done, the plugin will then "behave" in a better way, ie, on Homebridge restart it will:

* go through the `ignoredDevices` list and remove any corresponding accessories found in the cache
* ignore any discovered devices that are in the `ignoredDevices` list
* add accessories back if they are no longer in the `ignoredDevices` list

**Comment by bwp91** [comment]
@idcrook if you're on the Homebridge Discord I would be very willing to message you on there to clarify things better!

**Comment by idcrook** [comment]
can you describe what you think putting a device in the `ignoredDevices` means, from a user's point of view?

**Comment by bwp91** [comment]
Sure, basically that the plugin will behave as if the device does not exist.
